_posts/2018-02-12-continuous-releases.md

```diff
@@ -0,0 +1,107 @@
+
+This meant that releases were somewhat infrequent, and features spent a long time in master before they were available to `pip install`.
+The pace of development picked up in 2017 -- partly as new maintainers arrived, and partly work for David's PhD -- and we wanted to be able to release more frequently.
+We decided to automate the entire release process.
+
+Now, when you create a pull request that changes the Hypothesis code -- anything that gets installed by pip, not the tests or docs -- you have to include a `RELEASE.rst` file, which describes your change.
+Here's an example from [a recent pull request][recent]:
+
+    RELEASE_TYPE: patch
+
+    This release changes the way in which Hypothesis tries to shrink the size of
+    examples. It probably won't have much impact, but might make shrinking faster
+    in some cases. It is unlikely but not impossible that it will change the
+    resulting examples.
+
+The first line says whether this is a major, minor, or patch release (using [semantic versioning][semver]).
+The rest is a description of the changes in your patch.
+
+We have a test in CI that checks for this file -- any change to the core code needs a release file, even [fixing a typo][typo].
+If you need a release file but haven't written one, the tests fail and your pull request won't be merged.
+
+Once you've written a release file and the pull request is merged into master, and after all the other tests have passed, our CI uses this file to create a new release.
+
+First, it works out the new version number, and updates it in [version.py][version.py].
+Then it copies the release description into the changelog, including the new version number and the current date.
+For example:
+
+    --------------------
+    3.44.25 - 2018-02-05
+    --------------------
+
+    This release changes the way in which Hypothesis tries to shrink the size of
+    examples. It probably won't have much impact, but might make shrinking faster
+    in some cases. It is unlikely but not impossible that it will change the
+    resulting examples.
+
+These two changes are saved as a new commit, and that commit gets tagged as the new release.
+The tag and the commit are pushed to GitHub, and then CI builds a new package and publishes it to PyPI.
+
+So with no very little extra work, every code change triggers a new release, and it's usually available within half an hour of merging the pull request.
+
+[recent]: https://github.com/HypothesisWorks/hypothesis-python/pull/1101
+[semver]: https://semver.org/
+[typo]: https://github.com/HypothesisWorks/hypothesis-python/pull/1069
+[version.py]: https://github.com/HypothesisWorks/hypothesis-python/blob/master/src/hypothesis/version.py
+
+## Why bother?
+
+Moving to continuous releases has been amazing.
+
+Our releases happen much more quickly.
+Rather than waiting for somebody to do it manually, every patch is available as soon as our tests confirm it's okay.
```

Comment:
Another unmentioned advantage is that I *really* hated doing these manual releases! Collecting the changelog manually was really annoying, and we didn't really have a previously good system of change log automation.